scripts/flow/common.js

```diff
@@ -0,0 +1,69 @@
+import {weaponEffects} from "../weaponEffects.js";
+
+export class FlowInfo {
+	constructor (
+		{
+			sourceToken,
+			actionIdentifier,
+			targetTokens = null,
+			targetsMissed = new Set(),
+		}
+	) {
+		this.sourceToken = sourceToken;
+		this.actionIdentifier = actionIdentifier;
+		this.targetTokens = targetTokens;
+		this.targetsMissed = targetsMissed;
+	}
+}
+
+export function getTokenByIdOrActorId(id) {
+	let token = canvas.tokens.get(id);
+	if (!token) {
+		token = canvas.tokens.ownedTokens.filter(t => t.actor.id === id)?.[0];
+		if (!token) {
+			console.log(`Lancer Weapon FX | No token with id '${id}' found.`);
+			return null;
+		}
+	}
+	return token;
+}
+
+async function _executeMacroByName(
+	macroName,
+	sourceToken = {},
+	{
+		compendiumName = "lancer-weapon-fx.WeaponFX",
+		flowInfo = null,
+	} = {},
+) {
+	const pack = game.packs.get(compendiumName);
+	if (pack) {
+		const macro_data = (await pack.getDocuments()).find((i) => i.name === macroName)?.toObject();
+
+		if (macro_data) {
+			const temp_macro = new Macro(macro_data);
+
+			(temp_macro.flags["lancer-weapon-fx"] ||= {}).flowInfo = flowInfo;
```

Comment:
This line is the only functional change: new method for passing information into an executing macro.